net/golioth/Kconfig

```diff
@@ -285,17 +285,6 @@ config GOLIOTH_SYSTEM_CLIENT_RX_BUF_SIZE
 	  Size of receive buffer, which is used for reading data from network
 	  socket.
 
-config GOLIOTH_SYSTEM_SETTINGS
-	bool "Load credentials from persistent settings"
-	default y
-	depends on GOLIOTH_AUTH_METHOD_PSK
-	depends on SETTINGS
-	help
-	  When selected, Golioth credentials will be loaded from settings
-	  subsystem.
-
-if GOLIOTH_SYSTEM_SETTINGS
-
```

Comment:
Most notable in my mind is persistent storage for WiFI credentials. Currently we use the shell to set to those in the storage partition.

Reply:
Upstream Zephyr has now support for connection/network manager with some functionalities implemented like management of WiFi (or other medium type) interface. Such "generic" functionalities make most sense to be included as part of Zephyr, since those are not cloud-provider (Golioth) specific. We should make use of those as much as possible, so that:
a) we don't maintain too much utilities that are not directly related to Golioth as a platform
b) it makes less friction for users that want to integrate their existing (or under development) app into Golioth

In other words, it would be great to provide Golioth SDK users some utilities, even for downstream applications. But in those cases it makes more sense to upstream such feature and let it be improved by larger community. Development of such feature (if it is well-thought) should take the same time in both cases. So in most cases some example fits best in `samples/common/` IMO.

In case of PSK-ID/PSK TLS credentials, I think that the ideal situation would be to provide a "TLS credentials storage which utilizes Zephyr settings subsystem". The only thing that would be Golioth-specific is the TLS ID which is used to reference those credentials. This TLS ID is already configured in Kconfig. The rest of it make sense to be put in Zephyr upstream. We can create something temporarily here in SDK until a "proper" implementation is merged to Zephyr, so that we don't introduce any latency between development and being able to use it.

Settings shell is one example why above strategy with upstreaming would be beneficial. First a settings shell was created in Golioth SDK. Shortly after that a different implementation (but with similar functionality in mind) was merged to Zephyr. Then there was a conflict between Kconfig symbols. It was solved (by prepending "our" service with `GOLIOTH_` in Kconfig option name). Right now there is still a conflict in shell module, so either one is visible or the other, since both commands are named `settings`.

Reply:
I like this a lot! I agree, upstreaming is the way to go. 

I filed https://github.com/golioth/firmware-issue-tracker/issues/183 and https://github.com/golioth/firmware-issue-tracker/issues/184 for some of the work that you mentioned, feel free to edit/add/delete.

> We can create something temporarily here in SDK until a "proper" implementation is merged to Zephyr

Where should that live? Still `samples/common` or do we make another folder for things that are in the process of being upstreamed?